### PR TITLE
replace speakeasy with golang.org/x/crypto/ssh/terminal

### PR DESCRIPTION
**Pull request by avahowell**
See #2414

### CODE REVIEW

File: siac/walletcmd.go [hunk 1]
```diff
@@ -507,10 +517,12 @@ func walletunlockcmd() {
...
 		}
 	}
-	password, err := speakeasy.Ask("Wallet password: ")
+	fmt.Print("Wallet password: ")
+	passwordBytes, err := terminal.ReadPassword(0)
 	if err != nil {
 		die("Reading password failed:", err)
 	}
+	password = string(passwordBytes)
 	qs := fmt.Sprintf("encryptionpassword=%s&dictonary=%s", password, "english")
```
Comment (lukechampine) [inline siac/walletcmd.go]: I think the cast is actually unnecessary here; `%s` should format a `[]byte` as a `string`

**Comment by lukechampine** [comment]
Perhaps we should define a helper function to match the speakeasy API:

```go
func passwordPrompt(prompt string) (string, error) {
    fmt.Print(prompt)
    pw, err := terminal.ReadPassword(0)
    return string(pw), err
}
```

**Comment by avahowell** [comment]
Perhaps, but then we would likely have to duplicate that helper since this functionality is used in `siad` and `siac`. Probably not a big deal. Another thing to consider after doing some runtime testing is that `terminal` doesn't print a newline after reading the password, so to perfectly mirror the functionality the helper should be:

```go
func passwordPrompt(prompt string) (string, error) {
    fmt.Print(prompt)
    pw, err := terminal.ReadPassword(0)
    fmt.Println()
    return string(pw), err
}
```

**Comment by avahowell** [comment]
Updated with the suggested helper function

**Comment by lukechampine** [comment]
LGTM, but we need to update the `Makefile` now

**Comment by avahowell** [comment]
Rebasedd and updated the makefile.

**Pull request merged**